sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -116,9 +116,18 @@ def log_prob(
         # TODO Train exited here, entered after sampling?
         self.net.eval()
 
-        theta, x = utils.match_shapes_of_theta_and_x(
-            theta, x, self.x_o, norm_posterior_snpe
-        )
+        # Prepare theta to be a tensor with batch dim.
+        theta = ensure_theta_batched(torch.as_tensor(theta))
+
+        # Select and check x to condition on.
+        x = self._x_else_x_o(x)
+        x = atleast_2d(torch.as_tensor(x))
+        self._check_for_multiple_x(x)
+
+        # Repeat x in case of evaluation on multiple theta. This is needed below in
+        # nflows in order to have matching shapes of theta and context x when
+        # evaluating the NN.
```

Comment:
This comment should also go in the docstring of `_match_x_with_theta_batch_shape()`